src/UCTNode.cpp

```diff
@@ -273,6 +273,11 @@ UCTNode* UCTNode::uct_select_child(int color, bool is_root) {
     for (auto& child : m_children) {
         if (!child.active()) {
             continue;
+        } else {
+            child.inflate();
```

Comment:
This is going to increase the memory consumption by a huge amount and makes the whole UCTNodePointer optimization pointless.  If it isn't inflated you can skip the m_is_expanding check and assume it isn't being expanded.

Reply:
One more comment - this is going to conflict with the 'lockless UCTNode' implementation down there since it doesn't have the m_is_expanding anymore (it uses a separate enum)

Reply:
Thanks. I've updated the code to save memory.
For the conflicting, I think we have to live with it until one is merged, then we can do the rebase.
The change needs to be proved to be correct first anyway.